### ISSUE TITLE
Allow collections plugin to modify options object. Allow those options to be available to content/headers/triples/writers plugins

### ISSUE THREAD

**Issue by mhavey** [marklogic/marklogic-data-hub#809]
In the collections phase of harmonization, it would be nice to be able to add entries to the options.  My thinking is that collections is my one chance to decide the document scope of the harmonization job and I might want to pass along concomitant findings to follow-on modules. I tried setting options from collections module, but I notice they are not propagated.

**Comment by kosed** [comment]
+1
Having collector return something like a map would be even better, a special key could be reserved for the uri if necessary.
I have been in the situations where I created entities from different piece of information contained in multiple staging documents.
I ended up concatenating those pieces in the returned string value that is usually containing the uri.
A map (or other structure) would probably have been better option.
